lib/cisco_node_utils/interface.rb

```diff
@@ -903,6 +914,444 @@ def switchport_trunk_native_vlan=(val)
       end
     end
 
+    # Interface private vlan configuration properties
+    # private vlan mode (host, promisc, trunk promisc , trunk sec)
+    # private vlan mapping
+    # private vlan native vlan
+    # private vlan allow vlan
+
+    def cli_error_check(result)
+      # The NXOS interface private vlan cli does not raise an exception
+      # in some conditions and instead just displays a STDOUT error message
+      # thus NXAPI does not detect the failure.
+      # We must catch it by inspecting the "body" hash entry
+      # returned by NXAPI. This vlan cli behavior is unlikely to change.
+      # Check for messages that can be safely ignored.
+
+      errors = /(ERROR:|VLAN:|Warning:|Eth)/
+      return unless
+        result[1].is_a?(Hash) && errors.match(result[1]['body'].to_s)
+      # Split errors into a list, but keep the delimiter as part of the message.
+      error_list =
+        (result[1]['body'].split(errors) - ['']).each_slice(2).map(&:join)
+      error_list.each do |_msg|
+        fail result[1]['body']
+      end
+    end
+
+    def set_args_keys_default
+      keys = { name: @name }
+      @set_args = keys
+    end
+
+    def set_args_keys(hash={}) # rubocop:disable Style/AccessorMethodName
+      set_args_keys_default
+      @set_args = @set_args.merge!(hash) unless hash.empty?
+    end
+
+    def switchport_mode_private_vlan_host_lookup_string
+      case @name
+      when ETHERNET
+        return 'switchport_mode_private_vlan_host'
+      end
```

Comment:
1) Are you expecting more name types to match against? Just wondering why you used `case` instead of `if`
2) Why uppercase ETHERNET? We downcase it in the initialize method.